### PR TITLE
feat: reintroduced `genkit start`

### PR DESCRIPTION
**Pull request by pavelgj**
```
genkit start -- <command>
```

Ex:
```
genkit start -- tsx --watch src/index.ts
# or
genkit start -- go run main.go
```

This will bring up the dev UI in the foreground and run the code at the same time (setting `GENKIT_ENV=dev`).

Alternatively, if you want to run the Dev UI separately you can do:

```
genkit start
```

this will run ONLY the Dev UI (in the foreground), no runtime.

To run the code/runtime in dev mode (without the Dev UI):

```
genkit start --noui -- <command>
```

ex:

```
genkit start --noui -- tsx --watch src/index.ts
```

### CODE REVIEW

**Review by apascal07 — APPROVED** [review]

File: genkit-tools/cli/src/commands/start.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+interface RunOptions {
+  noui?: boolean;
+  port?: string;
+}
+
+/** Command to run code in dev mode and/or the Dev UI. */
+export const start = new Command('start')
+  .description('runs a command in genkit dev mode')
+  .option('-n, --noui', 'do not start the dev ui', false)
```
Comment (pavelgj) [inline genkit-tools/cli/src/commands/start.ts:32]:
```suggestion
  .option('-n, --noui', 'do not start the Dev UI', false)
```

File: genkit-tools/cli/src/commands/start.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  port?: string;
+}
+
+/** Command to run code in dev mode and/or the Dev UI. */
+export const start = new Command('start')
+  .description('runs a command in genkit dev mode')
+  .option('-n, --noui', 'do not start the Dev UI', false)
+  .option('-p, --port', 'port for the dev ui')
+  .action(async (options: RunOptions) => {
+    console.log(start.opts(), start.args);
```
Comment (pavelgj) [inline genkit-tools/cli/src/commands/start.ts:35]:
```suggestion
```

File: genkit-tools/cli/src/commands/start.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import { startManager } from '../utils/manager-utils';
+
+interface RunOptions {
+  noui?: boolean;
+  port?: string;
+}
+
+/** Command to run code in dev mode and/or the Dev UI. */
+export const start = new Command('start')
+  .description('runs a command in genkit dev mode')
```
Comment (pavelgj) [inline genkit-tools/cli/src/commands/start.ts:31]:
```suggestion
  .description('runs a command in Genkit dev mode')
```

File: genkit-tools/cli/src/commands/start.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  .option('-n, --noui', 'do not start the dev ui', false)
+  .option('-p, --port', 'port for the dev ui')
```
Comment (pavelgj) [inline genkit-tools/cli/src/commands/start.ts:33]:
```suggestion
  .option('-p, --port', 'port for the Dev UI')
```

File: genkit-tools/cli/src/commands/start.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pavelgj) [reply genkit-tools/cli/src/commands/start.ts]: Respect applied!

**Pull request merged**